backend/app/views/spree/admin/orders/_shipment_address_form.html.erb

```diff
@@ -0,0 +1,111 @@
+      </p>
+    </div>
+    <div class="col-md-6">
+      <p class="form-group">
+        <%= form.label :address2, Spree.t(:street_address_2) %>
+        <%= form.text_field :address2, class: 'form-control', value: address.address2 %>
+      </p>
+    </div>
+  </div>
+
+  <% if Spree::Config[:company] %>
+    <p class="form-group">
+      <%= form.label :company, Spree.t(:company) %>
+      <%= form.text_field :company, class: 'form-control', value: address.company %>
+    </p>
+  <% end %>
+
+  <p class="form-group">
+    <%= form.label :city, Spree.t(:city) %>
+    <%= form.text_field :city, class: 'form-control', value: address.city %>
+  </p>
+
+  <p class="form-group">
+    <%= form.label :zipcode, Spree.t(:zip) %>
+    <%= form.text_field :zipcode, class: "form-control", value: address.zipcode %>
+  </p>
+
+  <div class="row no-padding-bottom js-country-states">
+    <div class="col-md-6">
+      <p class="form-group js-country">
+        <%= form.label :country_id, Spree.t(:country) %>
+        <%= select :address, :country_id, options_for_select(available_countries.collect {|p| [ p.name, p.id ] }, address.country_id), {}, {class: 'select2'} %>
+      </p>
+    </div>
+    <% if Spree::Config[:address_requires_state] %>
+      <div class="col-md-6">
+        <p class="form-group js-state" id="<%="#{address_id}state" %>">
+          <% have_states = address.country.present? && !address.country.states.empty? %>
+          <%= form.label :state, Spree.t(:state) %>
+
+          <% state_elements = [
+             select(:address,
+                    :state_id,
+                    options_for_select(have_states ? address.country.states.collect {|p| [ p.name, p.id ] } : [], address.state_id),
+                    {},
+                    {include_blank: true,
+                     disabled: !have_states,
+                     class: have_states ? 'select2' : 'select2 hidden'}) +
+             form.text_field(:state_name,
+                                class: !have_states ? 'form-control' : 'form-control hidden',
+                                disabled: have_states,
+                                value: address.state_name)
+             ].join.gsub('"', "'").gsub("\n", "")
+          %>
+          <%= javascript_tag do -%>
+            $('#<%="#{address_id}state" %>').append("<%== state_elements %>");
+          <% end %>
+        </p>
+      </div>
+    <% end %>
+  </div>
+
+  <p class="form-group">
+    <%= form.label :phone, Spree.t(:phone) %>
+    <%= form.phone_field :phone, class: "form-control", value: address.phone %>
+  </p>
+
+  <% if Spree::Config[:alternative_shipping_phone] %>
+    <p class="form-group">
+      <%= form.label :alternative_phone, Spree.t(:alternative_phone) %>
+      <%= form.phone_field :alternative_phone, class: 'form-control', value: address.alternative_phone %>
+    </p>
+  <% end %>
+
+  <% other_shipments = @order.shipments.where.not(id: shipment.id, state: 'shipped') %>
```

Comment:
@simmsy mentioned desire to add new shipment states as well soon so we may want to make this state check explicitly for pending or ready rather than not shipped.